### DISCUSSION TITLE
Question about RIS delay

### DISCUSSION THREAD

**Discussion by fanyimeng0** [NVlabs/sionna#566] [Q&A] [answered]
Hello, I would like to know how you handle the delay in RIS. I noticed that scene.compute_paths doesn't seem to return the delay for the TX-RIS-RX path. Could you provide some insights on how to address this issue?

Thank you!

**Comment by jhoydis** [comment]
Hi @fanyimeng0,

Every path computed by `scene.compute_paths` has a delay and a complex amplitude, including paths involving RIS. I am hence not sure to understand your question. Could you provide a code snippet that explains the problem you encounter?

**Reply by fanyimeng0** [reply]
[issue.zip](https://github.com/user-attachments/files/16942135/issue.zip)
In this Jupyter notebook, I noticed that the dimensions of a and tau do not match correctly

**Reply by fanyimeng0** [reply]
Sionna version is 0.18.0, and TensorFlow version is 2.13.1.

**Reply by jhoydis** [reply, marked as answer]
I can reproduce the bug. There seems to be something wrong in the `paths.cir()` method when the flag `cluster_ris_paths` is enabled.
You should be able to get the delays for all individual paths by deactivating the flag :
``
a, tau = paths.cir(cluster_ris_paths=False)
``
This will be fixed in the next release.

**Reply by fanyimeng0** [reply]
Thank you for your quick response.
